README.md

```diff
@@ -1,191 +1,154 @@
+| `RESOURCE_BASE_URL` | `base-theme` | `https://open-learning-course-data-rc.s3.amazonaws.com/` | A base URL to prefix the rendered path to resources with |
+| `WWW_HUGO_CONFIG_PATH` | `www` | `/path/to/ocw-hugo-projects/ocw-www/config.yaml` | A path to the `ocw-www` Hugo configuration file |
+| `COURSE_HUGO_CONFIG_PATH` | `course` | `/path/to/ocw-hugo-projects/ocw-course/config.yaml` | A path to the `ocw-course` Hugo configuration file |
+| `WWW_CONTENT_PATH` | `www` | `/path/to/ocw-content-rc/ocw-www` | A path to a Hugo site that will be rendered when running `npm run start:www` |
+| `COURSE_CONTENT_PATH` | `course` | `/path/to/ocw-content-rc/` | A path to a base folder containing `ocw-course` type Hugo sites |
+| `OCW_TEST_COURSE` | `course` | `18.06-spring-2010` | The name of a folder in `COURSE_CONTENT_PATH` containing a Hugo site that will be rendered when running `npm run start:course` |
+| `OCW_IMPORT_STARTER_SLUG` | `www` | `ocw-course` | When generating "New Courses" cards on the home page, the `ocw-studio` API is queried using `OCW_STUDIO_BASE_URL`.  This value determines the `type` used in the query string against the API |
+| `COURSE_BASE_URL` | N/A | `http://localhost:3000/courses` | Used in `build_all_courses.sh`, this is the `--baseUrl` argument passed to each course build iterated by the script |
+| `VERBOSE` | N/A | `0` | Used in `build_all_courses.sh`, if set to `1` this will print verbose output from the course builds to the console |
+| `DOWNLOAD` | N/A | `1` | Used in `npm run start:course`, if set to `0` this will not download course data from S3 and instead source `OCW_TEST_COURSE` from the specified `OCW_TO_HUGO_OUTPUT_DIR`. If not specified, it will default to 1 and try to download data from S3. |
+
+### Running ocw-www
+
+To run `ocw-www` locally for working on the `www` theme:
+
+ - Clone the `ocw-www` content repo: https://github.mit.edu/ocw-content-rc/ocw-www
+ - Clone `ocw-hugo-projects` to obtain the relevant configuration file: https://github.com/mitocw/ocw-hugo-projects
+ - Set the following environment variables in your `.env` file, replacing `/path/to` with your path to the repos indicated:
+   - `WWW_HUGO_CONFIG_PATH=/path/to/ocw-hugo-projects/ocw-www/config.yaml`
+   - `WWW_CONTENT_PATH=/path/to/ocw-content-rc/ocw-www/`
+ - Optionally set these environment variables as well, depending on the functionality you need to work on:
+   - `SEARCH_API_URL=https://discussions-rc.odl.mit.edu/api/v0/search/` (for testing search functionality)
+   - `OCW_STUDIO_BASE_URL=http://ocw-studio-rc.odl.mit.edu/` (for testing `ocw-studio` API functionality, such as the "new courses" section rendered by `www/layouts/partials/home_course_cards.html`)
+   - `RESOURCE_BASE_URL=https://open-learning-course-data-rc.s3.amazonaws.com/` (if you need to test the loading of resources from S3 or some other CDN)
```

Comment:
what would you think about automating these setup steps? I would _love_ to have a `dev.sh` script or similar which automatically handled the happy path for all these things so that testing out a styling change locally was really straightforward (we could then provide overrides via env variables or something for things which might need to change)

Reply:
You mean like some test content to spin up `www` or `course` sites without any external content?  I think that could be done.  We'd have to create and maintain a "baseline" set of content for each theme but I think that's probably a good idea anyway and could be used for some basic assertion of the output as well.  I'll work on that.

Reply:
I don't know that test content is necessary - but just something where there is a zero-config way to get the site going by running a single script, perhaps just by automating cloning the repos mentioned here, setting the right environment variables, etc. One of the things I've felt has been "friction-ey" about spinning up the ocw-hugo-themes for development is that the dev setup takes a number of steps, so if you haven't been working on it for a bit it is a bit error prone and can end up taking up a chunk of time if you do something slightly wrong in a difficult-to-detect way. So if there are, as outlined here in the README, a set of relatively concrete, repeatable steps for spinning up the ocw-www site, it would be great I think to automate those if we can.

Kind of like the equivalent of how for running ocw-studio locally in development we really only support using `docker-compose`, and, aside from environment variables which are somewhat messy to configure, bringing up the whole application and it's environment is just `docker-compose build && docker-compose up`. It would be great to just be able to do something like `./ocw-www-dev.sh` and have something similar happen.